### DISCUSSION TITLE
请问 IEditor 中 insertSnippet 的参数如何获得？

### DISCUSSION THREAD

**Discussion by zhuzeyu22** [opensumi/core#2426] [Q&A]
我想在文本新建的时候初始化一段默认代码
调用 BrowserEditorContribution 中的 registerEditorFeature 得到了 IEditorFeatureRegistry 接口，调用 registerEditorFeatureContribution 的 contribute 获得 IEditor 调用 insertSnippet ，里面的 ranges 和 opts 怎么获得/填写 ？

**Comment by Ricbet** [comment]
ranges 是你定义补全的这段代码要写入的位置和范围，类型定义在 `@opensumi/ide-core-common` 中
```typescript
import { IRange } from '@opensumi/ide-core-common';
```

opts 的两个参数表示说围绕当前的补全编辑操作的 撤销/回退 行为，完整的类型定义你可以看一下 `packages/types/vscode/typings/vscode.editor.d.ts` 这个文件

**Reply by zhuzeyu22** [reply]
IEditorFeatureContribution 的接口实现还需要实现 provideEditorOptionsForUri 方法， 这个方法我不想实现，怎么能跳过去？

**Reply by Ricbet** [reply]
这个方法是可选的，可以不用实现
